tools/make-dnsmasq-conf.yml

```diff
@@ -0,0 +1,142 @@
+      delegate_to: localhost
+      run_once: true
+  gather_facts: false
+
+- name: generate hostvars[myvars] for all hosts
+  hosts: all
+  gather_facts: false
+  tasks:
+    - name: 'set myvars'
+      set_fact:
+        myvars:
+          fqdn: "{{ inventory_hostname }}"
+          bmc_ip: "{{ ipmi }}"
+          bmc_mac: "{{ bmc }}"
+          boot_ip: "{{ hostvars[inventory_hostname][maas_boot_ip_var] }}"
+          boot_mac: "{{ hostvars[inventory_hostname][maas_boot_mac_var] }}"
+          domain: "{{ inventory_hostname | regex_replace('^[^.]+\\.', '') }}"
+          shortname: "{{ inventory_hostname | regex_replace('\\..*', '') }}"
+          domain_tag: "{{ inventory_hostname.split('.')[1:-4] | join('-') }}"
+      tags: always
+
+
+- name: generate files from hostvars[myvars]
+  hosts: all
+  gather_facts: false
+  vars:
+    ipmi_file: "../dnsmasq.conf/ipmi.conf"
+    ipmi_domain: "ipmi.ceph.tuc.ibm.com"
+
+  tasks:
+    #- name: get myvars
+    #  set_fact:
+    #    myvars: "{{ hostvars[inventory_hostname].myvars }}"
+
+    - name: build {{ ipmi_file }}
+      copy:
+        dest: "{{ ipmi_file }}"
+        content: |
+          ###
+          ### {{ ipmi_domain }}
+          ###
+          {% for host in ansible_play_hosts %}
+          {% set data = hostvars[host]['myvars'] %}
+          dhcp-host=set:ipmi,{{ data.bmc_mac }},{{ data.bmc_ip }},{{ data.shortname }}.{{ ipmi_domain }}
+          {% endfor %}
+          {% for host in ansible_play_hosts %}
+          {% set data = hostvars[host]['myvars'] %}
+          host-address={{ data.bmc_ip }},{{ data.shortname }}.{{ ipmi_domain }}
+          {% endfor %}
+          {% for vlan in maas_networking[0]['vlans'] %}
+          {% set data = hostvars[ansible_play_hosts[0]].myvars %}
+          {% for subnet in vlan['subnets'] %}
+          {% if data.bmc_ip | ansible.utils.ipaddr(subnet.cidr) %}
+          dhcp-range={{ subnet.gateway | ipmath(5) }},static,{{ subnet.cidr | ipaddr("netmask") }},{{ subnet.cidr | ipaddr("broadcast") }}
+          dhcp-option=tag:ipmi,option:router,{{ subnet.gateway }}
+          {% endif %}
+          {% endfor %}
+          {% endfor %}
+
+      delegate_to: localhost
+      run_once: true
+      tags: ipmi
+
+    # create a map from domain_tag to a list of hosts, so that we can loop
+    # across all the domain_tags
+    - name: Build map of domain_tag →> list of hosts
+      set_fact:
+        hosts_by_domain_tag: >-
+          {{
+            hosts_by_domain_tag | default({}) | combine({
+              (hostvars[item]['myvars']['domain_tag']): (
+                (hosts_by_domain_tag | default({})).get(hostvars[item]['myvars']['domain_tag'], []) + [item]
+              )
+            })
+          }}
+      loop: "{{ ansible_play_hosts }}"
+      loop_control:
+        loop_var: item
+      delegate_to: localhost
+      run_once: true
+
+
+    # maas_networking variable:
+    # maas_networking['tucson-qe']['vlans'][]['subnets'][]['cidr']
```

Comment:
Is this still here on purpose?

Reply:
sort of.  it's there so you can understand the code below without having to refer to group_vars